.github/workflows/ci.yml

```diff
@@ -0,0 +1,60 @@
+name: GitHub CI
+
+on:
+  push:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+      - '*.yml'
+  pull_request:
+    paths-ignore:
+      - '.github/*'
+      - '.github/*_TEMPLATE/**'
+      - '*.md'
+      - '*.yml'
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout nxdk
+        uses: actions/checkout@v2
+        with:
+          repository: XboxDev/nxdk
+          submodules: recursive
+      - name: Set xbox_kernel_test_suite dir
+        id: xkts
+        run: echo "::set-output name=dir::projects/xbox_kernel_test_suite"
+      - name: Checkout xbox_kernel_test_suite
+        uses: actions/checkout@v2
+        with:
+          path: ${{ steps.xkts.outputs.dir }}
+      - name: Install dependencies
+        run: |
+          sudo apt-get -y update
+          sudo apt-get -y install lld llvm
+      - name: Build
+        working-directory: ${{ steps.xkts.outputs.dir }}
+        run: make -j$(nproc)
+      - name: Set artifact+release parameters
+        id: parameters
+        run: |
+          export file=default.xbe
+          echo "::set-output name=name::$file"
+          echo "::set-output name=path::${{ steps.xkts.outputs.dir }}/bin/$file"
+          echo "::set-output name=tag::$(date +'%Y%m%d%H%M%S')-${GITHUB_SHA:0:7}"
+      - uses: actions/upload-artifact@v2
+        with:
+          name: ${{ steps.parameters.outputs.name }}
+          path: ${{ steps.parameters.outputs.path }}
+          if-no-files-found: error
+      - name: Create release
+        if: |
+          github.event.action != 'pull_request' &&
+          github.ref == 'refs/heads/master'     &&
+          github.repository == 'Cxbx-Reloaded/xbox_kernel_test_suite'
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: gh release create ${{ steps.parameters.outputs.tag }} ${{ steps.parameters.outputs.path }} --target $GITHUB_REF -R $GITHUB_REPOSITORY
```

Comment:
https://github.com/Cxbx-Reloaded/xbox_kernel_test_suite/pull/69#discussion_r638211036
Like you said `GITHUB_REF` is a branch name. By doing this way, you are going to introduce an issue. If multiple pull requests merged at same time will tagged be at same commit as latest branch's. Please replace `GITHUB_REF` to `GITHUB_SHA` where it should make a release on specific commit. (This is intended design for create-release's action.)

As for `-R $GITHUB_REPOSITORY`, for what reason do we need it?

Reply:
`-R $GITHUB_REPOSITORY` is needed because otherwise it tries to create a release on the main repository which is problematic for testing on forks.

Reply:
Base on gh's repository info, `GITHUB_REPOSITORY` is not used by default. Instead, it is checking for `GH_REPO` environment. I recommend to use the environment set than pass in argument. The reason I preferred in environment setter is to remain noticable. Until official github's gh tool proceed integrate support for `GITHUB_REPOSITORY` environment.

Reply:
Using `env:` doesn't work for some reason:
```
        env:
          GH_REPO: ${{ env.GITHUB_REPOSITORY }}
```
https://github.com/Margen67/Test/runs/2868128291#step:2:6

Reply:
I made a test to be sure if result is correct. Turns out all `GITHUB_` prefix is intended to be run from run command and excluded from env context unless provided one within workflow file. Proper solution is to use 
```
        env:
          GH_REPO: ${{ github.repository }}
```

Reply:
Correction: It tries to create a release on XboxDev/nxdk because it's the workspace. Under normal circumstances overriding the repo isn't necessary.